Inlining a library with namespaced imports/exports inlines export statements

Kinda related to #266

_(tested on https://github.com/timocov/dts-bundle-generator/pull/270, v8.1.2 produces a different error)_

**Bug report**

**Input code**

```ts
import { z } from 'zod';

export const obj = z.object({
    comets: z.number().min(1),
});
```

**Expected output**

All of used `zod` (without export statements) + generated types from input file.

**Actual output**

Fails with the following errors:

```
src/index.d.ts(919,15): error TS2307: Cannot find module './errors' or its corresponding type declarations.
src/index.d.ts(920,15): error TS2307: Cannot find module './helpers/parseUtil' or its corresponding type declarations.
src/index.d.ts(921,15): error TS2307: Cannot find module './helpers/typeAliases' or its corresponding type declarations.
src/index.d.ts(922,15): error TS2307: Cannot find module './helpers/util' or its corresponding type declarations.
src/index.d.ts(923,15): error TS2307: Cannot find module './types' or its corresponding type declarations.
src/index.d.ts(924,15): error TS2307: Cannot find module './ZodError' or its corresponding type declarations.
```

The generated output contains the following statements:

```
export * from "./errors";
export * from "./helpers/parseUtil";
export * from "./helpers/typeAliases";
export * from "./helpers/util";
export * from "./types";
export * from "./ZodError";
```

**Additional context**

```bash
npx dts-bundle-generator --external-inlines zod -- src/index.ts
```

cc @yovanoc

--- a/src/bundle-generator.ts
+++ b/src/bundle-generator.ts
@@ -207,7 +207,18 @@ export function generateDtsBundle(entries: readonly EntryPointConfig[], options:
 				for (const declaration of getDeclarationsForExportedValues(exportAssignment)) {
 					let exportedDeclarations: readonly ts.Statement[] = [];
 
-					if (ts.isModuleDeclaration(declaration)) {
+					if (ts.isExportDeclaration(exportAssignment) && ts.isSourceFile(declaration)) {
+						const referencedModule = getReferencedModuleInfo(exportAssignment, criteria, typeChecker);
+						if (referencedModule !== null) {
+							if (visitedModules.has(referencedModule.fileName)) {
+								continue;
+							}
+
+							visitedModules.add(referencedModule.fileName);
+						}
+
+						exportedDeclarations = declaration.statements;
+					} else if (ts.isModuleDeclaration(declaration)) {
 						if (declaration.body !== undefined && ts.isModuleBlock(declaration.body)) {
 							const referencedModule = getReferencedModuleInfo(declaration, criteria, typeChecker);
 							if (referencedModule !== null) {
@@ -258,7 +269,7 @@ export function generateDtsBundle(entries: readonly EntryPointConfig[], options:
 					}
 
 					if (ts.isExportDeclaration(statement)) {
-						if (!currentModule.isExternal) {
+						if (currentModule.type === ModuleType.ShouldBeInlined) {
 							continue;
 						}
 

--- a/src/helpers/typescript.ts
+++ b/src/helpers/typescript.ts
@@ -541,7 +541,13 @@ export function getClosestSourceFileLikeNode(node: ts.Node): ts.SourceFile | ts.
 	return ts.isSourceFile(node) ? node : node.parent;
 }
 
-export type NodeWithReferencedModule = ts.ExportDeclaration | ts.ModuleDeclaration | ts.ImportTypeNode | ts.ImportEqualsDeclaration | ts.ImportDeclaration;
+export type NodeWithReferencedModule =
+	| ts.ExportDeclaration
+	| ts.ImportDeclaration
+	| ts.ImportEqualsDeclaration
+	| ts.ImportTypeNode
+	| ts.ModuleDeclaration
+;
 
 export function resolveReferencedModule(node: NodeWithReferencedModule, typeChecker: ts.TypeChecker): ts.SourceFile | ts.ModuleDeclaration | null {
 	let moduleName: ts.Expression | ts.LiteralTypeNode | undefined;

--- a/tests/e2e/test-cases/re-export-in-node_modules/config.ts
+++ b/tests/e2e/test-cases/re-export-in-node_modules/config.ts
@@ -1,0 +1,9 @@
+import { TestCaseConfig } from '../test-case-config';
+
+const config: TestCaseConfig = {
+	libraries: {
+		inlinedLibraries: ['package-with-re-exports'],
+	},
+};
+
+export = config;

--- a/tests/e2e/test-cases/re-export-in-node_modules/input.ts
+++ b/tests/e2e/test-cases/re-export-in-node_modules/input.ts
@@ -1,0 +1,3 @@
+import { Barfoo, Foobar, content } from 'package-with-re-exports';
+
+export type Baz = Barfoo | Foobar | content.Foobar;

--- a/tests/e2e/test-cases/re-export-in-node_modules/output.d.ts
+++ b/tests/e2e/test-cases/re-export-in-node_modules/output.d.ts
@@ -1,0 +1,7 @@
+export interface Foobar {
+}
+export interface Barfoo {
+}
+export type Baz = Barfoo | Foobar | Foobar;
+
+export {};

